token-swap/program/src/curve/stable.rs

```diff
@@ -194,14 +203,36 @@ impl CurveCalculator for StableCurve {
         trade_direction: TradeDirection,
```

Comment:
`TradeDirection` is not the same thing as `deposit` vs `withdrawal`, and instead tells which token you'll be adding / removing.  The `RoundDirection` kind of tells you if you're doing a deposit or withdrawal, but isn't very clear for that.  You'll want to change `RoundDirection` to give that information instead, which means updating some of the code for the other curves.  You can change it to `LiquidityProviderOperation` with names `Withdraw` and `Deposit`